api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -740,7 +751,249 @@ public void shouldTestPurgeProgramAttributeType() throws Exception {
                 int totalAttributeTypes = pws.getAllProgramAttributeTypes().size();
 		pws.purgeProgramAttributeType(programAttributeType);
 		assertEquals((totalAttributeTypes - 1), pws.getAllProgramAttributeTypes().size());
-}
+	}
+	
+	
+	@Test
+	public void getPrograms_shouldTestGetPrograms() {
+		List<Program> programs = pws.getPrograms("MAL");
+		List<Program> programs1 = pws.getPrograms("PR");
+		assertEquals(programs.size(), 1);
+		assertEquals(programs1.size(), 3);
+	}
+	
+	@Test
+	public void retireProgram_shouldTestRetireProgram() {
+		Program programTest = pws.getAllPrograms().get(0);
+		Program program = pws.retireProgram(programTest, "expired");
+		for (ProgramWorkflow workflow : program.getWorkflows()) {
+			assertEquals(workflow.getRetired(), true);
+			for (ProgramWorkflowState state : workflow.getStates()) {
+				assertEquals(state.getRetired(), true);
+			}
+		}
+		
+	}
+	
+	@Test
+	public void unretireProgram_shouldTestUnretireProgram() {
```

Comment:
same as above comment for test case name

Reply:
I meant to modify the test case name like below. Leaving comment is not recommended to describe the test case. Please remove the comment.
```suggestion
	public void unretireProgram_shouldSetRetireFalseForWorkflowsAndWorkflowStates() {
```